### ISSUE TITLE
[BUG]:  Vulkan no longer works after update

### ISSUE THREAD

**Issue by Kagewani123** [Vita3K/Vita3K#3300]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

v.0.2.0 3606

### Description of Issue

Vulkan doesn't work anymore after most recent update. Despite my specs, it worked perfectly before the recent update.

### Expected Behavior

It's meant to work with Vulkan.

### Reproduction Steps

Try to play anything with Vulkan.

### Log File

[22:42:59.658] |I| [config::init_config]: backend-renderer: Vulkan
[22:42:59.659] |I| [config::init_config]: log-level: Trace
[22:42:59.866] |I| [SDL_main]: Vita3K v0.2.0 3606-b8ef1dd7
[22:42:59.868] |I| [SDL_main]: OS: Microsoft Windows 10  (build 19045)
[22:42:59.875] |I| [SDL_main]: CPU: Intel(R) Core(TM) i3-1005G1 CPU @ 1.20GHz | 4 Threads | 1.19 GHz
[22:42:59.876] |I| [SDL_main]: Available ram memory: 7959 MiB
[22:42:59.885] |I| [app::init]: Base path: C:\Users\dadsm\Downloads\vita3k-0-2-0-build-3594\
[22:42:59.885] |I| [app::init]: User pref path: C:\Users\dadsm\AppData\Roaming\Vita3K\Vita3K\
[22:43:00.134] |I| [renderer::vulkan::VKState::create]: Vulkan device: Intel(R) UHD Graphics
[22:43:00.135] |I| [renderer::vulkan::VKState::create]: Driver version: 100.9365
[22:43:00.135] |I| [renderer::vulkan::VKState::create]: Memory mapping is enabled
[22:43:00.175] |I| [renderer::vulkan::ScreenRenderer::setup]: Present mode: Fifo
[22:43:01.116] |I| [compat::update_app_compat_db]: Applications compatibility database is up to date.
[22:45:18.746] |I| [renderer::vulkan::VKState::late_init]: Using shader interlock for accurate framebuffer fetch emulation
[22:45:18.748] |I| [renderer::vulkan::PipelineCache::set_async_compilation]: Enabling asynchronous pipeline compilation with 1 threads
[22:45:18.752] |I| [load_app_impl]: cpu-backend: Dynarmic
[22:45:18.752] |I| [load_app_impl]: CPU Optimisation state: true
[22:45:18.752] |I| [load_app_impl]: ngs state: true
[22:45:18.752] |I| [load_app_impl]: Resolution multiplier: 1
[22:45:18.752] |I| [load_app_impl]: modules mode: Automatic
[22:45:18.752] |I| [load_app_impl]: Title: Soul Sacrifice™ Delta
[22:45:18.753] |I| [load_app_impl]: Serial: PCSA00152
[22:45:18.753] |I| [load_app_impl]: Version: 1.00
[22:45:18.753] |I| [load_app_impl]: Category: gd
[22:45:18.756] |I| [load_module]: Loading module "app0:eboot.bin"
[22:45:19.195] |I| [load_self]: app0:eboot.bin: Skipping special segment 0x6fffff01...
[22:45:19.196] |I| [load_self]: Loaded module segment 0 @ [0x81000000 - 0x81C92BB0 / 0x81000000] (size: 0x00C92BB0) of module app0:eboot.bin
[22:45:19.196] |I| [load_self]: Loaded module segment 1 @ [0x81C93000 - 0x81E1BF0C / 0x81C93000] (size: 0x00188F0C) of module app0:eboot.bin
[22:45:19.197] |I| [load_self]: Linking SELF app0:eboot.bin...
[22:45:19.198] |D| [load_var_exports]:  NID 0x6c2224ba (NID_MODULE_INFO) at 0x81b2858c
[22:45:19.198] |D| [load_var_exports]:  NID 0x70fba1e7 (SCE_PROC_PARAMS) at 0x81c9a77c
[22:45:19.200] |D| [load_var_imports]:  NID NOT FOUND 0x36878958 (UNRECOGNISED) at 0x81b2aaf8, setting to stub value 0xdeadbeef
[22:45:19.201] |D| [load_var_imports]:  NID NOT FOUND 0x3ce6109d (UNRECOGNISED) at 0x81b2a96c, setting to stub value 0xdeadbeef
[22:45:19.202] |D| [load_var_imports]:  NID NOT FOUND 0x5d8c1282 (UNRECOGNISED) at 0x81b2accc, setting to stub value 0xdeadbeef
[22:45:19.202] |D| [load_var_imports]:  NID NOT FOUND 0xd662e07c (UNRECOGNISED) at 0x81b2aab4, setting to stub value 0xdeadbeef
[22:45:19.203] |D| [load_var_imports]:  NID NOT FOUND 0xddf9bb09 (UNRECOGNISED) at 0x81b2ac14, setting to stub value 0xdeadbeef
[22:45:19.204] |D| [load_var_imports]:  NID NOT FOUND 0x53f02a18 (UNRECOGNISED) at 0x81b2aa20, setting to stub value 0xdeadbeef
[22:45:19.205] |D| [load_var_imports]:  NID NOT FOUND 0x9c449444 (UNRECOGNISED) at 0x81b2ab90, setting to stub value 0xdeadbeef
[22:45:19.206] |D| [load_var_imports]:  NID NOT FOUND 0xd9478dcf (UNRECOGNISED) at 0x81b2ab2c, setting to stub value 0xdeadbeef
[22:45:19.207] |I| [load_module]: Module efg (at "app0:eboot.bin") loaded
[22:45:19.209] |I| [load_app_impl]: Main executable efg (eboot.bin) loaded
[22:45:19.210] |I| [load_module]: Loading module "app0:sce_module/libc.suprx"
[22:45:19.223] |I| [load_self]: app0:sce_module/libc.suprx: Skipping special segment 0x6fffff01...
[22:45:19.224] |I| [load_self]: Loaded module segment 0 @ [0x8000C000 - 0x800593BC / 0x81000000] (size: 0x0004D3BC) of module app0:sce_module/libc.suprx
[22:45:19.225] |I| [load_self]: Loaded module segment 1 @ [0x8005A000 - 0x8005C6F0 / 0x8104E000] (size: 0x000026F0) of module app0:sce_module/libc.suprx
[22:45:19.226] |I| [load_self]: Linking SELF app0:sce_module/libc.suprx...
[22:45:19.227] |D| [load_var_exports]: Found previously not found variable. nid:0x36878958, new_entry_point:0x80051200
[22:45:19.228] |D| [load_var_exports]: Found previously not found variable. nid:0x3ce6109d, new_entry_point:0x8004ff6c
[22:45:19.228] |D| [load_var_exports]: Found previously not found variable. nid:0x5d8c1282, new_entry_point:0x8005a4c0
[22:45:19.229] |D| [load_var_exports]: Found previously not found variable. nid:0xd662e07c, new_entry_point:0x80050ff8
[22:45:19.229] |D| [load_var_exports]: Found previously not found variable. nid:0xddf9bb09, new_entry_point:0x8005a5c0
[22:45:19.231] |D| [load_var_exports]: Found previously not found variable. nid:0x53f02a18, new_entry_point:0x80058318
[22:45:19.232] |D| [load_var_exports]: Found previously not found variable. nid:0x9c449444, new_entry_point:0x80055c6c
[22:45:19.232] |D| [load_var_exports]: Found previously not found variable. nid:0xd9478dcf, new_entry_point:0x8005be58
[22:45:19.233] |D| [load_var_exports]:  NID 0x936c8a78 (SYSLYB) at 0x8004f164
[22:45:19.234] |D| [load_var_exports]:  NID 0x6c2224ba (NID_MODULE_INFO) at 0x800450fc
[22:45:19.234] |I| [load_module]: Module SceLibc (at "app0:sce_module/libc.suprx") loaded
[22:45:19.235] |I| [load_module]: Loading module "app0:sce_module/libfios2.suprx"
[22:45:19.244] |I| [load_self]: app0:sce_module/libfios2.suprx: Skipping special segment 0x6fffff01...
[22:45:19.245] |I| [load_self]: Loaded module segment 0 @ [0x8005D000 - 0x8008812C / 0x81000000] (size: 0x0002B12C) of module app0:sce_module/libfios2.suprx
[22:45:19.246] |I| [load_self]: Loaded module segment 1 @ [0x80004000 - 0x8000923D / 0x8102C000] (size: 0x0000523D) of module app0:sce_module/libfios2.suprx
[22:45:19.247] |I| [load_self]: Linking SELF app0:sce_module/libfios2.suprx...
[22:45:19.248] |D| [load_var_exports]:  NID 0x936c8a78 (SYSLYB) at 0x80088034
[22:45:19.249] |D| [load_var_exports]:  NID 0x6c2224ba (NID_MODULE_INFO) at 0x8007f548
[22:45:19.249] |I| [load_module]: Module SceLibFios2 (at "app0:sce_module/libfios2.suprx") loaded
[22:45:19.250] |I| [load_module]: Loading module "vs0:sys/external/libSceFt2.suprx"
[22:45:19.261] |I| [load_self]: Loaded module segment 0 @ [0x80089000 - 0x800D9F04 / 0x81000000] (size: 0x00050F04) of module vs0:sys/external/libSceFt2.suprx
[22:45:19.262] |I| [load_self]: Loaded module segment 1 @ [0x8000A000 - 0x8000A2F4 / 0x81051000] (size: 0x000002F4) of module vs0:sys/external/libSceFt2.suprx
[22:45:19.263] |I| [load_self]: Linking SELF vs0:sys/external/libSceFt2.suprx...
[22:45:19.264] |D| [load_var_exports]:  NID 0x936c8a78 (SYSLYB) at 0x800c57dc
[22:45:19.264] |D| [load_var_exports]:  NID 0x6c2224ba (NID_MODULE_INFO) at 0x800c4f10
[22:45:19.265] |I| [load_module]: Module SceLibft2 (at "vs0:sys/external/libSceFt2.suprx") loaded
[22:45:19.266] |I| [load_module]: Loading module "vs0:sys/external/libpvf.suprx"
[22:45:19.269] |I| [load_self]: Loaded module segment 0 @ [0x800DA000 - 0x800E9F14 / 0x81000000] (size: 0x0000FF14) of module vs0:sys/external/libpvf.suprx
[22:45:19.269] |I| [load_self]: Loaded module segment 1 @ [0x8000B000 - 0x8000B008 / 0x81010000] (size: 0x00000008) of module vs0:sys/external/libpvf.suprx
[22:45:19.271] |I| [load_self]: Linking SELF vs0:sys/external/libpvf.suprx...
[22:45:19.271] |D| [load_var_exports]:  NID 0x936c8a78 (SYSLYB) at 0x800e8710
[22:45:19.271] |D| [load_var_exports]:  NID 0x6c2224ba (NID_MODULE_INFO) at 0x800e8300
[22:45:19.272] |I| [load_module]: Module SceLibPvf (at "vs0:sys/external/libpvf.suprx") loaded
[22:45:19.293] |D| [start_module]: Running module_start of library: SceLibc at address 0x8000c809
[22:45:19.529] |T| [log_import_call]: [LLE] TID: 7   FUNC: 0xdae8d60f time at 0x81b27e6c
[22:45:19.530] |T| [log_import_call]: [LLE] TID: 7   FUNC: 0x6dc1f0d8 memset at 0x81b27c4c
[22:45:19.536] |T| [open_file]: sceKernelGetStdin: Opening terminal tty0:
[22:45:19.536] |T| [open_file]: sceKernelGetStdout: Opening terminal tty0:
[22:45:19.537] |T| [open_file]: sceKernelGetStderr: Opening terminal tty0:
[22:45:19.555] |I| [start_module]: Module SceLibc (at "app0:sce_module/libc.suprx") module_start returned 0x0
[22:45:19.559] |D| [start_module]: Running module_start of library: SceLibFios2 at address 0x8005d001
[22:45:19.608] |W| [export_sceKernelGetProcessId::<lambda_1>::operator ()]: Stubbed sceKernelGetProcessId import called. (pid: 1)
[22:45:19.614] |I| [start_module]: Module SceLibFios2 (at "app0:sce_module/libfios2.suprx") module_start returned 0x0
[22:45:19.620] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0xbfe02b3a __cxa_set_dso_handle_main at 0x81b27e0c
[22:45:19.621] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x54f87eac _sceLdTlsRegisterModuleInfo at 0x81b27bcc
[22:45:19.628] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0xedc939e1 __aeabi_atexit at 0x81b27edc
[22:45:19.631] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0xd0310e31 __cxa_guard_acquire at 0x81b27e4c
[22:45:19.632] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x4ed1056f __cxa_guard_release at 0x81b27b9c
[22:45:19.642] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x7205bfdb memcpy at 0x81b27c5c
[22:45:20.009] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x8aecc873 strlen at 0x81b27cfc
[22:45:20.263] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x5e60b2b3 _ZNSt8ios_base4InitC1Ev at 0x81b2808c
[22:45:20.264] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0xabb11cf9 _ZNSt6_WinitC1Ev at 0x81b280ec
[22:45:20.523] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x7449b359 sprintf at 0x81b27c6c
[22:45:20.534] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x789215c3 sceFiosFileGetSizeSync at 0x81b262cc
[22:45:20.537] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x1c488b32 sceFiosFileReadSync at 0x81b2627c
[22:45:20.538] |I| [export_sceIoOpen]: Opening file: app0:resource/boot_config.ini
[22:45:20.539] |T| [open_file]: sceIoOpen: Opening file app0:resource/boot_config.ini (ux0:/app/PCSA00152/resource/boot_config.ini), fd: 0x3
[22:45:20.542] |T| [close_file]: sceIoClose: Closing file fd: 0x3
[22:45:20.545] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x1b58fa3b strcmp at 0x81b27b0c
[22:45:20.547] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x2f3e5b16 memchr at 0x81b27b4c
[22:45:20.547] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0x7747f6d7 memcmp at 0x81b27c7c
[22:45:20.552] |T| [log_import_call]: [LLE] TID: 6   FUNC: 0xe4299dcb strncmp at 0x81b27e9c
[22:45:20.556] |W| [export_sceAppUtilInit::<lambda_1>::operator ()]: Unimplemented sceAppUtilInit import called.
[22:45:20.679] |C| [logging::init::<lambda_2>::operator ()]: Unhandled C++ exception. vk::Device::allocateMemory: ErrorInvalidOpaqueCaptureAddress

### System Configuration

CPU: Intel(R) Core(TM) i3-1005G1 CPU @ 1.20GHz | 4 Threads | 1.19 GHz
GPU/Driver: Intel(R) UHD Graphics
RAM: 7959 MiB
OS: Microsoft Windows 10  (build 19045)

**Comment by Kagewani123** [comment]
Just as I thought, no fucking help at all.

**Comment by EXtremeExploit** [comment]
sorry but we have a life, someone with the knowledge required to help with your problem will reply to your whenever that person feels like it because this project is free and no one has an obligation to do anything , its all from the faint of everyone's heart.
Have a nice day :D

**Comment by Kagewani123** [comment]
Sorry, but I have been very stressed about it.

**Comment by Kagewani123** [comment]
Anyone?

**Comment by Kagewani123** [comment]
Still getting this, even after the new update. Vulkan is broken for me.

**Comment by Kagewani123** [comment]
Please I just want some fucking help on how to fix this I've tried everything

**Comment by Zangetsu38** [comment]
exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx

**Comment by Kagewani123** [comment]
> exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx

It was originally working fine for me

**Comment by Zangetsu38** [comment]
> > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> 
> It was originally working fine for me

if you try with last build working that keep works ?

**Comment by Kagewani123** [comment]
> > > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> > 
> > 
> > It was originally working fine for me
> 
> if you try with last build working that keep works ?

I tried that, still doing it

**Comment by Zangetsu38** [comment]
> > > > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> > > 
> > > 
> > > It was originally working fine for me
> > 
> > 
> > if you try with last build working that keep works ?
> 
> I tried that, still doing it

You want said, old build keep works ?

**Comment by nn9dev** [comment]
> > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> 
> It was originally working fine for me

If you try the older build (the one you claim worked), does the emulator still work as intended?

**Comment by Kagewani123** [comment]
> > > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> > 
> > 
> > It was originally working fine for me
> 
> If you try the older build (the one you claim worked), does the emulator still work as intended?

No it doesn't

**Comment by Kagewani123** [comment]
> > > exept if is issue on Intel(R) UHD Graphics only 🤔 works fine for me in rtx
> > 
> > 
> > It was originally working fine for me
> 
> If you try the older build (the one you claim worked), does the emulator still work as intended?

No matter what I do, I get that error when I try to start a game with Vulkan

**Comment by Kagewani123** [comment]
Could it be related to this?
![image_2024-05-28_002420084](https://github.com/Vita3K/Vita3K/assets/146495905/3523c666-8e06-45cf-8f58-816e799bde17)

**Comment by Kagewani123** [comment]
Also, do I need Vulkan Run Time Libraries?

**Comment by BoB1990** [comment]
Looks like minimal version of Vulkan required is 1.2 now.
At least it is what I was able to find in [Magnum documentation](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1Vk.html#a11c0202b1193a81ca4b449671d48c8c5a95053955629087a3b16bcfac5b3767ce):
![image](https://github.com/Vita3K/Vita3K/assets/4854878/fe5f52fa-8085-4fb6-a065-b083b2b41dd5)

@Zangetsu38 Could you, please, check with the team and confirm?
If it is the case then there should be appropriate information in Vita3K FAQ docs:
https://github.com/Vita3K/Vita3K/wiki/FAQ#what-are-the-system-requirements-i-need-to-run-it
At the moment article states "GPU: An OpenGL 4.4 and/or **Vulkan _1.0_** capable GPU"

**Comment by nn9dev** [comment]
> Looks like minimal version of Vulkan required is 1.2 now.

I personally am unable to find any occurrences of the function that's supposed to *cause* this error, `vkGetBufferOpaqueCaptureAddress` (which is in turn supposed to be called via [some black magic I don't yet understand](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkBufferOpaqueCaptureAddressCreateInfo.html#_c_specification), I still haven't gotten around to learning Vulkan...), in any of ***Vita3K's*** code. Obviously it's there in the external folder, but this is just headers and such for Vulkan, which don't necessarily get used. Anyhow, for Vulkan 1.0 (and 1.1 ?) this functionality is exposed via the [VK_KHR_buffer_device_address](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_buffer_device_address.html) extension, which is an ***optional*** extension for driver makers/creators/programmers, which does in turn still make the requirement Vulkan 1.0, *technically*. The page for the extension linked above explicitly states that Vulkan 1.2 brought the extensions into the core API (aka "promoted").

Also, Vita3K should also detect if this extension is present and disable such memory mapping functions, as seen [here](https://github.com/Vita3K/Vita3K/blob/master/vita3k/renderer/src/vulkan/renderer.cpp#L369) and [here](https://github.com/Vita3K/Vita3K/blob/master/vita3k/renderer/src/vulkan/renderer.cpp#L412). Thus, this should end up being a non-issue...

And JUST to put the nail in the coffin, [the PDF release notes](https://downloadmirror.intel.com/824226/ReleaseNotes_101.2128_WHQL.pdf) for the [latest graphics driver](https://www.intel.com/content/www/us/en/download/776137/intel-7th-10th-gen-processor-graphics-windows.html) for OP's chip (i3-1005G1) explicitly states that it's driver is Vulkan ***1.3*** compliant. As far as I can tell, the "Driver version" being reported by Vita3K in the log posted refers to driver version [27.20.100.9365](https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=xgd4r), a driver released in 2021, to which I cannot immediately find a release notes for.

> Also, do I need Vulkan Run Time Libraries?

For this application, no.

> Could it be related to this?

I have no idea what you've posted a screenshot of, but whatever it is, I don't believe it's related to this project...

> > If you try the older build (the one you claim worked), does the emulator still work as intended?
> 
> No it doesn't

This would... logically... lead someone to believe that it might not be our fault...? The build that you say broke Vulkan doesn't actually touch anything in the renderer code. Closest I can see is [this change](https://github.com/Vita3K/Vita3K/commit/584e1f172ffdf2183081d899a065a004daab40cc#diff-a52cb2865b7acc62d6f91d89d87824516a3c5fd5260fd5ea862dfcc40d30362f) in the shader translator that shouldn't have broken anything...

ALSO, just so I wasn't doing the "talk shit, get hit" thing, I did JUST test Soul Sacrifice Delta on my laptop and it boots up just fine, so obviously this is not some core issue/plague that is just going unnoticed.

My recommendation? OP, [update your graphics drivers](https://www.intel.com/content/www/us/en/download/776137/intel-7th-10th-gen-processor-graphics-windows.html) and report back. Else, wait for someone with a far more know-how and far less time on their hands than I to take a peek at it. Might be something obvious I'm missing, who knows?

**Comment by BoB1990** [comment]
Hi @nn9dev,
Thanks for reply, glad you with us.
Yesterday I faced the same behavior that leaded to inability to use Vulkan on my Lenovo ThinkBook 14 IIL (Intel Core i7-1065G7, Intel Iris Plus Graphics 940).
And, indeed, updating my graphics driver resolved same error.
But in my mind it proves that Vulkan 1.2 is needed, at least in case of Intel 10th Gen Processor Graphics.

Thank you for providing link to Vulkan documentation, I will try to find out more details regarding this issue tomorrow.
In another issue ticket #2748 user states:
> I use old v0.1.5.2786 is OK,but use after v0.1.6.3000 is error.
Maybe this information will help to figure this out tomorrow.
Thanks again and have a nice weekend)

**Comment by Macdu** [comment]
As @nn9dev stated, no part of Vita3K is using the OpaqueCaptureAddress Vulkan functions/capabilities.
Also:
- The minimum requirement for Vita3K has always been Vulkan 1.0 with a single required extension (storage buffers) that is supported by basically all GPUs supporting Vulkan.
- You found randomly that Opaque Capture adresses require Vulkan 1.2 to be supported. That's not true: first a device only supporting Vulkan 1.0 can still support it with the proper extensions. Second Vulkan 1.2 only allows without extensions to ask if the feature is supported (but it may not be): my AMD GPU has Vulkan 1.3 and (unfortunately) does not support opaque capture addresses.

So then why do you have an error related to opaque capture addresses when running Vita3K? The reason is that although Vita3K does not use it, third-party tools that insert themselves between Vita3K and Vulkan can use them. In particular, replaying tools (like Renderdoc, nvidia nsight, gfxreconstruct...) will automatically use opaque capture addresses (even if the gpu does not support it) when trying to capture an app using buffer device addresses (which Vita3K uses). So I would say with 95% certainty that you just have a third party tool trying to record what Vita3K renders, and this is not supported by your GPU.

TLDR: You have a third-party tool pretty badly configured always getting ready to capture any Vulkan application, this causes the crash. This also has nothing to do with Vita3K itself.
